context-propagation/src/main/java/io/micrometer/context/ContextSnapshot.java

```diff
@@ -205,6 +205,105 @@ static ContextSnapshot captureFrom(Object context, Predicate<Object> keyPredicat
         return DefaultContextSnapshot.captureFromContext(keyPredicate, registry, context, null);
     }
 
+    /**
+     * Return a new {@code Runnable} that sets {@code ThreadLocal} values from the
+     * snapshot around the invocation of the given {@code Runnable}.
+     * @param runnable the runnable to instrument
+     */
+    static Runnable instrument(Runnable runnable) {
```

Comment:
I predict the name to be a source of confusion when discussing actual instrumentation (e.g. metrics / tracing) in the context propagation realm vs the mechanics used here. I'm not sure whether I have a better name, but perhaps
`ContextSnapshot.around(Runnable)` or `ContextSnapshot.preserve(Runnable)` and the following overrides would be more intuitive that the methods deal with the actual `ContextSnapshot` content instead of performing any instrumentation.